asm/f64/asm_test.go

```diff
@@ -65,3 +74,119 @@ func nonStridedWrite(x []float64, inc int) bool {
 	}
 	return false
 }
+
+// guardVector copies the source vector (vec) into a new slice with guards.
+// Guards guarded[:gdLn] and guarded[len-gdLn:] will be filled with sigil value gdVal.
+func guardVector(vec []float64, gdVal float64, gdLn int) (guarded []float64) {
+	guarded = make([]float64, len(vec)+gdLn*2)
+	copy(guarded[gdLn:], vec)
+	for i := 0; i < gdLn; i++ {
+		guarded[i] = gdVal
+		guarded[len(guarded)-1-i] = gdVal
+	}
+	return guarded
+}
+
+// isValidGuard will test for violated guards, generated by guardVector.
+func isValidGuard(vec []float64, gdVal float64, gdLn int) bool {
+	for i := 0; i < gdLn; i++ {
+		if !same(vec[i], gdVal) || vec[len(vec)-1-i] != gdVal {
```

Comment:
I linked the commit in which this was fixed.  It's from a commit on gonum/internal#54, so it's fixed on that PR.